e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/add-to-an-existing-payment.spec.js

```diff
@@ -149,4 +163,93 @@ context(`${PDC_TEAMS.PDC_RECONCILE} users can add fee records to existing paymen
       pages.utilisationReportAddToAnExistingPaymentPage.errorSummary().contains('Select a payment to add the fee or fees to');
     });
   });
+
+  it('should add the selected fees to the payments and update the fee record status after clicking the continue button', () => {
+    cy.task(NODE_TASKS.REMOVE_ALL_UTILISATION_REPORTS_FROM_DB);
+    cy.task(NODE_TASKS.REMOVE_ALL_PAYMENTS_FROM_DB);
+    cy.task(NODE_TASKS.INSERT_UTILISATION_REPORTS_INTO_DB, [report]);
+    cy.task(NODE_TASKS.INSERT_FEE_RECORDS_INTO_DB, [firstFeeRecord, secondFeeRecord]);
+
+    navigateToAddToExistingPaymentScreenForFirstFeeRecord();
+
+    pages.utilisationReportAddToAnExistingPaymentPage.continueButton().click();
+
+    pages.utilisationReportPage.premiumPaymentsTab.getPaymentLink(PAYMENT_ID_ONE).should('contain', 'GBP 450.00');
+    pages.utilisationReportPage.premiumPaymentsTab.premiumPaymentsTable
+      .getStatus(FEE_RECORD_ID_ONE)
+      .invoke('text')
+      .then((text) => {
+        expect(text.trim()).to.equal('MATCH');
+      });
+  });
+
+  it('should redirect the user to the premium payments page when user clicks the back or cancel links and persist the selected fees', () => {
```

Comment:
Could we split this up into individual tests? ...Would this deviate from other "reports" E2E approaches?

At the moment there seems to be 3x things tested in a single `it('...)` - would make for easier reading and debugging if it's split up 

```js
describe('when a user clicks the back link', () => {
  it('should redirect the user to the premium payments page', () => { ... }

  it('should persist the selected fees', () => { ... }
});

describe('when a user clicks the cancel link', () => {
  it('should redirect the user to the premium payments page', () => { ... }

  it('should persist the selected fees', () => { ... }
});
```


and with using before/beforeEach etc,

Reply:
I agree that it would be nicer to split these two cases, but these were grouped due to the reasons Ruby mentions above. Happy to split if needed, leaving as is for now

Reply:
I really think these should be setup. It's not following best practices @RLavender98 , everything is bundled into a single "it" test (69 lines) and is doing a lot of things.

I would not worry about the time it takes to run, it is negligible, especially when compared to future debugging.

Reply:
You don't need to do a `beforeEach` for every such assertion, you could do a single `before` and then a `beforeEach` for subsequent assertions if required.

Reply:
> I would not worry about the time it takes to run, it is negligible, especially when compared to future debugging.

There is an argument in the other direction in that lots of `before`s and `beforeEach`s and nested describe blocks also makes it harder to debug as there are lots of different places set up can fail. 

That said, I don't feel too strongly about this, so happy for us to go with splitting it up further!

Reply:
@RLavender98 Thing is, when split up and things fail in the pipeline etc, it will tell you exactly which test and line failed.

If they're all bundled into a single "it", it will stop on the first thing that failed and only tell you about that one failure. Could end up pushing a fix, but it's only fixing 1 of 5 things failing for example, in the same single test